variable/subspan_view.cpp

```diff
@@ -29,14 +30,17 @@ auto make_empty_subspans(const ElementArrayView<T> &, const Dimensions &dims) {
 template <class T>
 auto make_subspans(T *base, const VariableConstView &indices,
                    const scipp::index stride) {
-  const auto &offset = indices.values<core::bucket_base::range_type>();
-  const auto len = offset.size();
-  std::vector<span<T>> spans;
-  spans.reserve(len);
-  for (scipp::index i = 0; i < len; ++i)
-    spans.emplace_back(scipp::span(base + stride * offset[i].first,
-                                   base + stride * offset[i].second));
-  return spans;
+  const auto spans = variable::transform<scipp::index_pair>(
```

Comment:
Not clear on why the implementation here needed to change - could you explain?

Reply:
Yes, there are two related reasons (and unfortunately this is not the first time we run into this):
- `offset` is of type `ElementArrayView`, which provides access to multi-dimensional arrays. Since it must handle multi-dimensional iteration, strides, and broadcasts its random access is very slow. Parts of this could have been alleviated by using iteration instead of `operator[]`. `transform` does this.
- `transform` gives us multi-threading for free.

Finally, as noted in the TODO, I want to refactor this anyway to direclty create the desired variable.